src/reader/mod.rs

```diff
@@ -127,11 +127,152 @@ macro_rules! configure_methods {
     };
 }
 
+macro_rules! read_event_impl {
```

Comment:
All of these macros should be moved into a separate module (see also my comment in `buffered_reader.rs`).

Reply:
The advantage of their definition there is that here is one usage place and those macros is very tied to functions where they used. I think it will be better to leave them here.